File: Python Leet/121.py
import math
from typing import List
import sys

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        total=0
        mini = sys.maxsize#float('inf')
        for i in prices:
            mini =min(mini,i)
            total =max(total,i-mini)
        
        return total

File: leetcode/121.cpp
// leetcode 121 best time to buy and sell stock

#include<bits/stdc++.h>
#include<algorithm>
#include<string>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int result = 0;
        int mini =INT_MAX;
        for(auto i : prices){
            mini =min(mini,i);
            result =max(result, i-mini);
        }
        return result;
    }
};
